jobsAdmin/JobsAdmin.go

```diff
@@ -301,15 +301,46 @@ func (ja *jobsAdmin) JobMgrEnsureExists(jobID common.JobID,
 
 // Same as JobMgrEnsureExists but with addition parameter of user logger.
 func (ja *jobsAdmin) JobMgrCreateWithLogger(jobID common.JobID,
-	level common.LogLevel, commandString string, logger common.ILoggerResetable) ste.IJobMgr {
+	level common.LogLevel, commandString string) ste.IJobMgr {
 
 	return ja.jobIDToJobMgr.EnsureExists(jobID,
 		func() ste.IJobMgr {
 			// Return existing or new IJobMgr to caller
-			return ste.NewJobMgr(ja.concurrency, jobID, ja.appCtx, ja.cpuMonitor, level, commandString, ja.logDir, ja.concurrencyTuner, ja.pacer, ja.slicePool, ja.cacheLimiter, ja.fileCountLimiter, logger)
+			return ste.NewJobMgr(ja.concurrency, jobID, nil, ja.cpuMonitor, level, commandString, ja.logDir, ja.concurrencyTuner, ja.pacer, ja.slicePool, ja.cacheLimiter, ja.fileCountLimiter, nil)
 		})
 }
 
+// JobMgrCleanup cleans all goroutine started by jobMgr.
```

Comment:
since it does more than this, let's update it

// JobMgrCleanup cleans up the jobMgr identified by the given jobId. It undoes what NewJobMgr() does, basically it does the following:
// 1. Stop all go routines started to process this job.
// 2. Release the memory allocated for this JobMgr instance.